File: PassTask/3.1P/Clock/Clock.cs
namespace ClockProgram
{
    public class Clock
    {
        // Fields
        private Counter _hour;
        private Counter _minute;
        private Counter _second;

        // Constructor
        public Clock()
        {
            _hour = new Counter("Hour");
            _minute = new Counter("Minute");
            _second = new Counter("Second");
        }

        // Methods
        public void Tick()
        {
            this.IncrementSecond();
        }

        public void Reset()
        {
            _second.Reset();
            _minute.Reset();
            _hour.Reset();
        }

        private void IncrementSecond()
        {
            _second.Increment();
            if (_second.Ticks == 60)
            {
                _second.Reset();
                this.IncrementMinute();
            }
        }
        private void IncrementMinute()
        {
            _minute.Increment();
            if (_minute.Ticks == 60)
            {
                _minute.Reset();
                this.IncrementHour();
            }
        }
        private void IncrementHour()
        {
            _hour.Increment();
            if (_hour.Ticks == 13)
            {
                _hour.Reset();
                _hour.Increment();
            }
        }

        public string GetTime()
        {
            return $"{this.Hour}:{this.Minute}:{this.Second}";
        }

        // Properties
        private string Hour
        {
            get
            {
                if (_hour.Ticks == 0) _hour.Increment();
                return _hour.Ticks.ToString("D2");
            }
        }
        private string Minute
        {
            get { return _minute.Ticks.ToString("D2"); }
        }
        private string Second
        {
            get { return _second.Ticks.ToString("D2"); }
        }
    }
}

File: PassTask/11.1P/Clock/clock.py
from counter import Counter


class Clock:
    def __init__(self):
        # Fields
        self._hour = Counter("Hour")
        self._minute = Counter("Minute")
        self._second = Counter("Second")

    # Methods
    def tick(self):
        self._increment_second()

    def reset(self):
        self._second.reset()
        self._minute.reset()
        self._hour.reset()

    def _increment_second(self):
        self._second.increment()
        if self._second.ticks == 60:
            self._second.reset()
            self._increment_minute()

    def _increment_minute(self):
        self._minute.increment()
        if self._minute.ticks == 60:
            self._minute.reset()
            self._increment_hour()

    def _increment_hour(self):
        self._hour.increment()
        if self._hour.ticks == 13:
            self._hour.reset()
            self._hour.increment()

    def get_time(self):
        return f"{self._hour_str}:{self._minute_str}:{self._second_str}"

    # Properties
    @property
    def _hour_str(self):
        if self._hour.ticks == 0:
            self._hour.increment()
        return f"{self._hour.ticks:02d}"

    @property
    def _minute_str(self):
        return f"{self._minute.ticks:02d}"

    @property
    def _second_str(self):
        return f"{self._second.ticks:02d}"
